### ISSUE TITLE
IP Reporting: Optional submission for "Met" and "No Progress" Outputs

### ISSUE THREAD

**Issue by kasfaw** [unicef/etools-partner-reporting-portal#259]
In IP Reporting, data entry on an indicator should be optional if the status of the output is either set to "Met" or "No Progress":

- [ ] backend allows submission of indicators in a progress report that have no data when a LL Output status is either met or no progress
- [ ] frontend allows submission of indicators in a progress report that have no data when a LL Output status is either met or no progress
- [ ] @rkrupinski  if progress report is not complete, a message should show saying: "You have not completed all required indicators for this progress report. Unless your Output status is Met or has No Progress, all indicator data needs to be completed."

**Comment by rkrupinski** [comment]
@kasfaw 

After digesting the requirements, here are some questions that remain:

1. Does this change only affect IP Reporting? Both IP & Cluster use the same API endpoint for submitting reports.
2. By "status" we mean the overall status set with radio buttons, correct?
3. Can we simply disable the submit button when submission criteria are not met? Server-side validations still stands for protection against *hackers*.
4. If the above is true, do we show a few words of explanation why it's disabled?
5. What does "have no data" mean exactly? Does it only take location data under consideration or includes narrative assessment too? Regarding location data, does "no data" mean all or just one location being "blank"?

cc @dfryta

**Comment by kasfaw** [comment]
@rkrupinski

1. Both cluster and ip reporting can use this logic 
2. Correct. overall_status
3. Makes sense!
4. That would be a nice to have but not a requirement. 
5. It means all location data not being fully completed (all cells within the required disaggregation table not filled in). 

**side note:** narrative field is required in ip reporting but optional in cluster reporting

**Comment by rkrupinski** [comment]
@kasfaw 

5. ...which can only be the case for newly generated reports since the UI won't let you submit "sparse" disaggregations.

**Comment by kasfaw** [comment]
exactly. Thought I would be explicit in my explanation :)

**Comment by rkrupinski** [comment]
@dfryta 

I need to know how the JSON response is gonna look in case of validation error. Now, when you try to submit a report (assuming fake data integrity bug), you get:

```json
{
  "errors": [
    { "message": "Indicator can be submitted only once." }
  ]
}
```

As there can be many messages, let's return `"You have not completed all required indicators for this progress report. Unless your Output status is Met or has No Progress, all indicator data needs to be completed."` from django, if applicable.

**Comment by dfryta** [comment]
So, I double check and `"Indicator can be submitted only once."` doesn't show when one of below occurs:
- submission_date is empty
- report_status is set to sent_back
I've already changed message to: `Indicator was already submitted. Please set status to SENT BACK.`

**Comment by kasfaw** [comment]
@dfryta can we tweak message to say: Indicator was already submitted. Your IMO will need to send it back for you to edit your submission.

**Comment by dfryta** [comment]
Sure :-)

**Comment by dfryta** [comment]
Done from my side.
